contracts/SinglePlayerCommit.sol

```diff
@@ -2,27 +2,35 @@
 pragma solidity ^0.6.10;
 pragma experimental ABIEncoderV2;
 
-import "@nomiclabs/buidler/console.sol";
+import { console } from "@nomiclabs/buidler/console.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
-import "@openzeppelin/contracts/math/SafeMath.sol";
+// import "@openzeppelin/contracts/math/SafeMath.sol";
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@chainlink/contracts/src/v0.6/ChainlinkClient.sol";
+//https://github.com/smartcontractkit/chainlink/issues/3153#issuecomment-655241638
+import "@chainlink/contracts/src/v0.6/vendor/SafeMath.sol";
 
-contract SinglePlayerCommit is Ownable {
+/// @title CommitPool single-player mode contract
+/// @notice Enables staking and validating performance. No social/pool functionality.
+contract SinglePlayerCommit is ChainlinkClient, Ownable {
     using SafeMath for uint256;
 
     /******************
     GLOBAL CONSTANTS
     ******************/
     IERC20 public token;
     uint256 BIGGEST_NUMBER = uint256(-1);
+    uint256 constant private ORACLE_PAYMENT = 1 * LINK;
 
     /***************
     DATA TYPES
     ***************/
+    /// @notice Activity as part of commitment with oracle address. E.g. "cycling" with ChainLink Strava node 
     struct Activity {
-        string name; // e.g. "cycling" with list scoped to activities supported by Strava
+        string name;
         address oracle;
         bool allowed;
+        bool exists;
```

Comment:
what does this give us that allowed doesn't?

Reply:
It can exist and not be allowed, leaving room for already active commitments to be resolved while migrating away from a certain activity by not letting it be available for newer commitments

Reply:
ah, good point!